glusterd2/brick/glusterfsd.go

```diff
@@ -131,3 +134,83 @@ func NewGlusterfsd(binfo Brickinfo) (*Glusterfsd, error) {
 func (b *Glusterfsd) ID() string {
 	return b.brickinfo.Path
 }
+
+// BrickStartMaxRetries represents maximum no. of attempts that will be made
+// to start brick processes in case of port clashes.
+const BrickStartMaxRetries = 3
+
+// Until https://review.gluster.org/#/c/16200/ gets into a release.
+// And this is fully safe too as no other well-known errno exists after 132
+
+//anotherEADDRINUSE is errno generated for rpc connection
+const anotherEADDRINUSE = syscall.Errno(0x9E) // 158
+
+func errorContainsErrno(err error, errno syscall.Errno) bool {
+	exiterr, ok := err.(*exec.ExitError)
+	if !ok {
+		return false
+	}
+	status, ok := exiterr.Sys().(syscall.WaitStatus)
+	if !ok {
+		return false
+	}
+	if status.ExitStatus() != int(errno) {
+		return false
+	}
+	return true
+}
+
+// These functions are used in vol-create, vol-expand and vol-shrink (TBD)
+
+//StartBrick starts glusterfsd process
+func (b Brickinfo) StartBrick() error {
+
+	brickDaemon, err := NewGlusterfsd(b)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < BrickStartMaxRetries; i++ {
+		err = daemon.Start(brickDaemon, true)
+		if err != nil {
+			if errorContainsErrno(err, syscall.EADDRINUSE) || errorContainsErrno(err, anotherEADDRINUSE) {
+				// Retry iff brick failed to start because of port being in use.
+				// Allow the previous instance to cleanup and exit
+				time.Sleep(1 * time.Second)
+			} else {
+				return err
+			}
+		} else {
+			break
+		}
+	}
+
+	return nil
+}
+
+//StopBrick will stop glusterfsd process
+func (b Brickinfo) StopBrick() error {
+
+	brickDaemon, err := NewGlusterfsd(b)
+	if err != nil {
+		return err
+	}
+
+	err = daemon.Stop(brickDaemon, true)
```

Comment:
nit: You can do ` return daemon.Stop(brickDaemon, true)` instead. Just a minor suggestion as `golint` tool run by [goreportcard](https://goreportcard.com/report/github.com/gluster/glusterd2) can catch it.